src/wasm-traversal.h

```diff
@@ -176,17 +176,33 @@ struct Walker : public VisitorType {
     setFunction(func);
     static_cast<SubType*>(this)->doWalkFunction(func);
     static_cast<SubType*>(this)->visitFunction(func);
+    setFunction(nullptr);
```

Comment:
It should have been done before I think just for cleanliness, yeah, but also I realized that if we walk the initializer expression of a segment, then we are walking an expression which does *not* have a function. So the current function should be null, to avoid confusion.